src/ProductConstructionService/ProductConstructionService.Api/Queue/PcsJobsProcessorStatus.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace ProductConstructionService.Api.Queue;
+
+public enum PcsJobsProcessorState
+{
+    ContinueWorking,
+    StoppedWorking,
+    FinishJobAndStop
+}
+
+public class PcsJobsProcessorStatus
+{
+    public PcsJobsProcessorStatus()
+    {
+        State = PcsJobsProcessorState.ContinueWorking;
+        Semaphore = new(1);
+    }
+
+    public void Reset()
+    {
+        State = PcsJobsProcessorState.ContinueWorking;
+        if (Semaphore.CurrentCount == 0)
+        {
+            Semaphore.Release();
+        }
+    }
+
+    public PcsJobsProcessorState State { get; set; }
+    public SemaphoreSlim Semaphore { get; }
```

Comment:
Can you explain how the semaphore is used? I think it should be an implementation detail on this class and the API of this class should instead provide some way to tell if we have access.

However, I don't follow how we're actually using this as it seems we just us this in the top-level only. But I don't understand the purpose.

Reply:
This semaphore is used to block the `PcsJobProcessor` when we stop it from processing new jobs. The `StatusController` it is releases it when we call the `start` endpoint

Reply:
I don't understand why we have the booleans but also the semaphore. Maybe you will have to explain it to me.

However it would still be better to hide the semaphore inside of the status class so that you can do something like `await _status.EnterAsync()` or something.

Inside the class, it can be a semaphore, or a monitor, or a lock, or a boolean, doesn't matter.